### PR TITLE
WIP: Optional values "the other way" and replace peer with peers

### PR DESCRIPTION
**Pull request by nsg**
# Proposed Changes

This commit is based on top of #15, I do not think this PR should be merged as-is, this is a breaking change that would break peoples installations. It implements the discussion in #14 and re-implements #15 in another way.

## Peers
If you like to go forward with this, or something similar I think we need to do one of 3 things:

a) Make it a breaking change, clearly communicate this to the users, or
b) Make some form of migration script that rewrites the configuration to the new format, or
c) Support both `peer` and `peers`

I initially tried to implement c, but it quickly got messy. I considered b as a better solution. In this PR i just did a :)

```
peers:
  - public_key: pUBlickey=
    pre_shared_key: ''
    endpoint: myserver.example.com:1234
    allowed_ips:
      - 10.0.0.0/24
    persistent_keep_alive: '25'
```
This is exactly the same syntax, just a list of possible multiple peers. You can inspect the commit that adds this functionality here: https://github.com/nsg/addon-wireguard-client/commit/698b4d3a7daebd74cbb621b92ace5f62e502e913

A few notes:
* Not sure if it's good to use internals like this https://github.com/nsg/addon-wireguard-client/commit/698b4d3a7daebd74cbb621b92ace5f62e502e913#diff-f5eeb82cf2a7d038d1baaace560daa12ee3d2ded417fc0f2d6d45278566ab68dR106
* I hope `bashio::jq` is a stable API to use https://github.com/nsg/addon-wireguard-client/commit/698b4d3a7daebd74cbb621b92ace5f62e502e913#diff-f5eeb82cf2a7d038d1baaace560daa12ee3d2ded417fc0f2d6d45278566ab68dR122 .. I found no other nice way to access the data from the loop otherwise.

## Optional values "the other way"

This is the alternative implementation of #15, instead of messing with empty variables this just makes the values optional via the built in functionality. This feels more like "The Home Assistant Way" but has the disadvantage that we wont get a nice sample configuration generated on first install. This change should be backward compatible.

See the commit here: https://github.com/nsg/addon-wireguard-client/commit/414f2f920c4a5000609262ff3e56edcf79ac6e9d

## To summarize

Feel free to close this, see this more as a PR with code samples and ideas ... I think https://github.com/nsg/addon-wireguard-client/commit/414f2f920c4a5000609262ff3e56edcf79ac6e9d could be an interesting alternative to #15 for example. If you feel you like to move forward with the idea of supporting multiple peers I would love to discuss what you think about the options a, b or c.

Oh, this code is not that well tested, it runs (for me) :grin: 

## Related Issues / PRs

* #15 
* #14

### CODE REVIEW

File: wireguard_client/config.json [hunk 1]
```diff
@@ -40,15 +40,12 @@
...
       },
-      "peer":{
+      "peers":[{
          "public_key": "",
          "pre_shared_key": "",
          "endpoint": "xxxxxx.duckdns.com",
-         "allowed_ips":[
-            "10.6.0.0/24"
-         ],
-         "persistent_keep_alive": "25"
```
Comment (bigmoby) [inline wireguard_client/config.json:50]: this param is very important to be keep

Comment (nsg) [reply wireguard_client/config.json:50]: It works just fine without `persistent_keep_alive`, if you only plan to make outbound connections it works. But you are correct that it's probably a good idea to make sure it's set.

**Comment by bigmoby** [comment]
Hi @nsg thank you so much for your contribute 🙏 
In these days I saw both PRs you left WIP and I get several hints from your code. I create the 0.1.9 version of this add-on, I hope you could enjoy it! 👍
Many thanks,
Bigmoby
